test/spec/controllers/notificationCtrlSpec.js

```diff
@@ -84,4 +85,111 @@ describe('Notification Controller', function() {
       expect(rootScope.openModal).to.not.be.called;
     });
   });
+
+  describe('User challenge won modal watch', function() {
```

Comment:
Note that `wonChallenge` is [not treated like a normal "achievement"](https://github.com/khipkin/habitrpg/blob/new_achievement_modal_service/website/public/js/controllers/notificationCtrl.js#L33).  If this is unintentional, I can fix it.